examples/api/test/widgets/implicit_animations/animated_positioned.0_test.dart

```diff
@@ -0,0 +1,56 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/widgets/implicit_animations/animated_positioned.0.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets(
+    'AnimatedPositioned animates on tap',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const example.AnimatedPositionedExampleApp(),
+      );
+
+      Positioned positioned = tester.widget(
+        find.descendant(
+          of: find.byType(AnimatedPositioned),
+          matching: find.byType(Positioned),
+        ),
+      );
+      expect(positioned.width, 50.0);
+      expect(positioned.height, 200.0);
+      expect(positioned.top, 150.0);
+
+      await tester.tap(find.text('Tap me'));
+      await tester.pump();
+
+      positioned = tester.widget(
+        find.descendant(
+          of: find.byType(AnimatedPositioned),
+          matching: find.byType(Positioned),
+        ),
+      );
+      expect(positioned.width, 50.0);
+      expect(positioned.height, 200.0);
+      expect(positioned.top, 150.0);
+
+      // Advance animation to the end by the 2-second duration specified in
+      // the example app.
+      await tester.pump(const Duration(seconds: 2));
```

Comment:
Nit 1: Maybe expose this duration from the example app and use it here.

Nit 2: Maybe pump half of the duration and check that the values are in between their start and end values, so we know that it animated.